utils/__tests__/AsyncIterableTask.test.ts

```diff
@@ -0,0 +1,85 @@
+import { number } from "fp-ts";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { forEach } from "lodash";
+import * as AI from "../AsyncIterableTask";
+
+async function* yield123() {
+  for (let i = 1; i <= 3; i++) {
+    yield i;
+  }
+}
+
+async function* yieldThrowError() {
+  for (let i = 1; i <= 3; i++) {
+    if (i < 3) yield i;
+    else throw Error("an Error");
+  }
+}
+
+describe("AsyncIterableTask", () => {
+  it("fold - should read all values", async () => {
+    const asyncIterable = yield123();
+    const asyncIterator = asyncIterable[Symbol.asyncIterator]();
+
+    const res = await pipe(
+      asyncIterator,
```

Comment:
If you explicit the `Iterable` and `Iterator` types you can check that you need to pass an `AsyncIterable` and not an `AsyncIterator`:

```suggestion
    const asyncIterable: AsyncIterable<number> = yield123();

    const res = await pipe(
      asyncIterable,
```

Reply:
Thanks! Done in [eb9bfeb](https://github.com/pagopa/io-functions-services-cache/pull/32/commits/eb9bfeba3948345afca296723fbccd21b409ce56)